site/lib/src/components/common/client/filtering.dart

```diff
@@ -0,0 +1,149 @@
+// Copyright 2025 The Dart Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:jaspr/dom.dart';
+import 'package:jaspr/jaspr.dart';
+import 'package:universal_web/web.dart' as web;
+
+class CheckboxFilterGroup<T> extends StatelessComponent {
+  const CheckboxFilterGroup({
+    required this.title,
+    required this.items,
+    required this.selectedItems,
+    required this.onToggle,
+    required this.labelProvider,
+    required this.idProvider,
+    super.key,
+  });
+
+  final String title;
+  final List<T> items;
+  final Set<T> selectedItems;
+  final void Function(T item, bool checked) onToggle;
+  final String Function(T item) labelProvider;
+  final String Function(T item) idProvider;
+
+  @override
+  Component build(BuildContext context) {
+    return div([
+      h4([.text(title)]),
+      ul([
+        for (final item in items)
+          li([
+            input(
+              type: InputType.checkbox,
+              attributes: {
+                'role': 'checkbox',
```

Comment:
No need to set the role to checkbox if its input type is already checkbox. Generally you can prefer the native handling if it's correct :)


```suggestion
```